sale_payment_term_interest/model/account_payment_term.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+#
+#
+#    Authors: Guewen Baconnier
+#    Copyright 2015 Camptocamp SA
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+import openerp.addons.decimal_precision as dp
+from openerp import models, fields, api
+from openerp.tools.float_utils import float_round as round
+
+
+class AccountPaymentTerm(models.Model):
+    _inherit = 'account.payment.term'
+
+    @api.multi
+    def compute_interest(self, value, date_ref=False):
+        if date_ref:
+            date_ref = fields.Date.from_string(date_ref)
+        else:
+            date_ref = datetime.today().date()
+
+        amount = value
+        result = []
+
+        lines_total = 0.0
+        precision_model = self.env['decimal.precision']
+        prec = precision_model.precision_get('Account')
+        for line in self.line_ids:
+            if line.value == 'fixed':
+                line_amount = round(line.value_amount, precision_digits=prec)
+            elif line.value == 'procent':
+                line_amount = round(value * line.value_amount,
+                                    precision_digits=prec)
+            elif line.value == 'balance':
+                line_amount = round(amount, prec)
```

Comment:
This code is the same than `account_payment_term.compute()` in the core account module. Since I want interests based on the amounts due for each term, I have to use the exact same computation than the `compute()` method.